Add docs for maintainers and enforce hugo version (#7)
Scientific-Python-Translations/scientific-python-translations.github.io: content/docs.md

is used to create a new branch including all (and only) commits from Crowdin's
branch for a particular language of interest. This script is useful because:

1. Crowdin will keep commits for all the available languages under translation
   in the same branch. However, we prefer to be able to create pull requests for
   one language at a time to keep Pull Request reviews smaller and easier to manage.

2. There are times when translations must be edited manually (outside of
   Crowdin) due to incorrect segmentation of the content into strings for
   translation. Crowdin's branch can only be edited through the Crowdin UI, and
   commits pushed to it outside of Crowdin will be lost when the branch is
   synced. By creating a new branch with only the commits we are interested in,
   we can bypass this limitation.

Ideally you will **not** have to run the `create_branch_for_language.sh` script
directly, but can include it as part of your website deployment process (see,
for example, [numpy/numpy.org#772](https://github.com/numpy/numpy.org/pull/772))

{{< admonition important >}}

> To prevent future conflicts with the GitHub/Crowdin integration, it is important
> that you set your translations to duplicate strings to be an exact match. To do
> this, navigate to your project's Settings in Crowdin, select Import and under
> "Source strings" -> "Duplicates" choose "Hide (regular detection)".

Reviewer: ```suggestion
To prevent future conflicts with the GitHub/Crowdin integration, it is important
that you configure Crowdin to have duplicate strings share the same translation.
To do this, navigate to your project's Settings in Crowdin, select Import and under
"Source strings" -> "Duplicates" choose "Hide (regular detection)".
```

Sorry, I think the language about an exact match is probably something I'd told you, but it turns out that what Crowdin means by an "exact match" is that the source strings match and the string IDs in the segmentation match. My understanding now is that "exact match"  is useful when you want translations to be shared for the same string across branches, but not the same string in different places within the same branch.

The suggestion here mirrors the language Crowdin uses to describe the feature.

is used to create a new branch including all (and only) commits from Crowdin's
branch for a particular language of interest. This script is useful because:

1. Crowdin will keep commits for all languages under translation in the same
   branch. However, we prefer to be able to create, review, and merge pull
   requests for one language at a time.

2. There are times when translations must be edited manually (outside of
   Crowdin) due to incorrect segmentation of the content into strings for
   translation. Crowdin's branch can only be edited through the Crowdin UI, and
   commits pushed to it outside of Crowdin will be lost when the branch is
   synced. By creating a new branch with only the commits we are interested in,
   we can bypass this limitation.

Ideally you will **not** have to run this script directly, but can include it as
part of your website deployment process (see, for example,
[numpy/numpy.org#772](https://github.com/numpy/numpy.org/pull/772))

{{< admonition important >}}
To prevent future conflicts with the GitHub/Crowdin integration, it is important

> that you set your translations to duplicate strings to be an exact match:

Reviewer: ```suggestion
To prevent future conflicts with the GitHub/Crowdin integration, it is important
that you the duplicate strings setting under *Settings*, *import*, *Duplicates* to
*Hide (regular detection) - all duplicates will share the same translation*.
```

Also. Does it make sense to include a screenshot?

![image](https://github.com/user-attachments/assets/073529d3-9bd2-4153-918c-3f427b0039b1)

Author: Yes - more importantly can you share how you got to this menu option? I tried going through the settings but could not find it

Author: Nevermind I found it!

> screenshot

Author: I can't find the correct setting in Crowdin, @steppi would you help me complete this?

`do-not-merge` label applied to it.

After translations for a language are completed and ready to be deployed, you
should:

1. Go to the repository corresponding to your project under
   https://github.com/Scientific-Python-Translations
2. Go to the Actions tab
3. Manually trigger the "Create translations PR" workflow, with the language
   code for your language of interest as input.

<center><img alt="Screenshot of the Actions tab from GitHub, with the Create translations PR workflow highlighted." src="../images/create_translations.png" width=800/></center>

<center><img alt="Screenshot of the Run workflow dialog from GitHub, with an input field labeled Crowding language code for the language of interest" src="../images/run_workflow.png" width=800/></center>

After these steps, a PR will be created to your website repo with the
translations for the language you selected (see
[numpy/numpy.org#774](https://github.com/numpy/numpy.org/pull/774) for an example.) This PR should be
merged when you are ready to publish the translations.

After merging the translations PR, you should **delete the corresponding branch** and

> re-create it from scratch. This will prevent conflicts with future translations.

Author: @steppi please confirm

> After merging the translations PR, you should **delete the corresponding branch** and
> re-create it from scratch. This will prevent conflicts with future translations.

Reviewer: ```suggestion
After merging the translations PR, the Crowdin service branch (by default, named `l10n_main`) will have merge conflicts with `main`. To fix this, delete the Crowdin service branch. Crowdin will automatically recreate the service branch with merge conflicts resolved. This same process can also be used to resolve merge conflicts if translations are updated outside of Crowdin. (Setting the duplicate strings setting to *Hide (regular detection) - all duplicates will share the same translation* makes Crowdin better able to resolve merge conflicts without un-translating strings).
```

Author: Thank you!

## Setting up a language switcher

> This work is in progress - follow (issue #) for details.

Author: We should create an issue to update folks on how this is going.

---

> title: "Documentation for maintainers"

Reviewer 2: Nit, but I generally prefer persona-first like Maintainers' docs

draft: false
---

This page details existing patterns for setting up and deploying translations
for project websites. As each project has their own set up, the patterns
described here may not fit exactly, but they will help you understand the
basic workflows.

At least some of the tasks described here rely on scripts from
https://github.com/Scientific-Python-Translations/automations/.

## Integration with Crowdin

> Crowdin provides GitHub integration tools to sync all translated content from

Reviewer 2: Link to Crowdin on first mention

the Crowdin web interface into project websites and vice-versa. However, a few
adaptations are needed.

> Given a GitHub repository which has been synced to Crowdin, the

Reviewer 2: ```suggestion
Once a GitHub repository has been synced to Crowdin, the
```

Make it easier to follow/read

[`create_branch_for_language.sh` script](https://github.com/Scientific-Python-Translations/automations/blob/main/scripts/create_branch_for_language.sh)
is used to create a new branch including all (and only) commits from Crowdin's
branch for a particular language of interest. This script is useful because:

> 1. Crowdin will keep commits for all languages under translation in the same
>    branch. However, we prefer to be able to create, review, and merge pull
>    requests for one language at a time.

Reviewer 2: ```suggestion
1. Crowdin will keep commits for all the available languages under translation in the same
   branch. However, we prefer to be able to create pull
   requests for one language at a time to keep Pull Request reviews smaller and easier to manage.
```

2. There are times when translations must be edited manually (outside of
   Crowdin) due to incorrect segmentation of the content into strings for
   translation. Crowdin's branch can only be edited through the Crowdin UI, and
   commits pushed to it outside of Crowdin will be lost when the branch is
   synced. By creating a new branch with only the commits we are interested in,
   we can bypass this limitation.

> Ideally you will **not** have to run this script directly, but can include it as

Reviewer 2: Which script? the create branch one?

Author: Yes - added a note

After these steps, a PR will be created to your website repo with the
translations for the language you selected (see
[numpy/numpy.org#774](https://github.com/numpy/numpy.org/pull/774) for an example.) This PR should be
merged when you are ready to publish the translations.

After merging the translations PR, the Crowdin service branch (by default, named `l10n_main`) will have merge conflicts with `main`. To fix this, delete the Crowdin service branch. Crowdin will automatically recreate the service branch with merge conflicts resolved. This same process can also be used to resolve merge conflicts if translations are updated outside of Crowdin. (Setting the duplicate strings setting to *Hide (regular detection) - all duplicates will share the same translation* makes Crowdin better able to resolve merge conflicts without un-translating strings).

## Setting up a language switcher

This work is in progress - follow (issue #) for details.

### Scientific Python Hugo Theme

### PyData Sphinx Theme

This work is in progress - follow [pydata/pydata-sphinx-theme#507](https://github.com/pydata/pydata-sphinx-theme/issues/507) for details.

## Known limitations

### Missing translations

> For items such as news items and release announcements, translations may not
> always be up to date. In this case, your project can decide what to do with
> these items (for example, keep them in english or hide them from the deployed
> site.)

Reviewer 2: ```suggestion
Translations may not always be up to date for items such as news items and release announcements. In this case, your project can decide what to do with
these items (for example, keep them in English or hide them from the deployed
site.)
```
